From: Thomas Gleixner <tglx@linutronix.de>
Date: Thu, 21 Jul 2011 21:05:33 +0200
Subject: [PATCH 076/347] net/wireless: Use WARN_ON_NORT()
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=2f2969b6e9d7684a5335400f044600660672d2dd

The softirq counter is meaningless on RT, so the check triggers a
false positive.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 net/mac80211/rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/rx.c b/net/mac80211/rx.c
index 3dc370ad23bf..96b2333496b5 100644
--- a/net/mac80211/rx.c
+++ b/net/mac80211/rx.c
@@ -4355,7 +4355,7 @@ void ieee80211_rx_napi(struct ieee80211_hw *hw, struct ieee80211_sta *pubsta,
 	struct ieee80211_supported_band *sband;
 	struct ieee80211_rx_status *status = IEEE80211_SKB_RXCB(skb);
 
-	WARN_ON_ONCE(softirq_count() == 0);
+	WARN_ON_ONCE_NONRT(softirq_count() == 0);
 
 	if (WARN_ON(status->band >= NUM_NL80211_BANDS))
 		goto drop;
